File: 240116/총점 비교/compare-total-points.py
n = int(input())

class student:

    def __init__(self,name,kor,math,eng):
        self.name = name
        self.kor = kor 
        self.math = math
        self.eng = eng

peo = [] 
for _ in range(n):
    name, kor, math, eng = map(str,input().split())
    kor = int(kor)
    math = int(math)
    eng = int(eng)

    peo.append(student(name,kor,math,eng))
peo.sort(key = lambda x: (x.kor+x.math+x.eng))

for i in range(n):
    print(peo[i].name,peo[i].kor,peo[i].math,peo[i].eng)

File: 240116/총점 비교/compare-total-points.java
import java.util.Scanner;
import java.util.Arrays;
class Student implements Comparable<Student>{
    String name;
    int kor,eng,math;

    public Student(String name,int kor,int eng,int math){
        this.name = name;
        this.kor = kor;
        this.eng = eng;
        this.math = math;
    }

    public int compareTo(Student student){
        return (this.kor+this.eng+this.math) - (student.kor+student.eng+student.math);
    }
}
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Student[] peo = new Student[n];

        for(int i = 0 ; i<n; i++){
            String name = sc.next();
            int kor = sc.nextInt();
            int eng = sc.nextInt();
            int math = sc.nextInt();

            peo[i] = new Student(name,kor,eng,math);
        }
        Arrays.sort(peo);

        for(int i = 0 ; i<n; i++){
            System.out.println(peo[i].name+" "+peo[i].kor+" "+peo[i].eng+" "+peo[i].math);
        }
    }
}